tf2_ros/src/buffer.cpp

```diff
@@ -246,17 +265,25 @@ Buffer::timerCallback(const TimerHandle & timer_handle,
                       TransformStampedFuture future,
                       TransformReadyCallback callback)
 {
+  bool timer_is_valid = false;
   tf2::TransformableRequestHandle request_handle;
   {
     std::lock_guard<std::mutex> lock(timer_to_request_map_mutex_);
-    request_handle = timer_to_request_map_.at(timer_handle);
+    auto timer_and_request_it = timer_to_request_map_.find(timer_handle);
+    timer_is_valid = (timer_to_request_map_.end() != timer_and_request_it);
+    if (timer_is_valid) {
+      request_handle = timer_and_request_it->second;
+    }
     timer_to_request_map_.erase(timer_handle);
+    timer_interface_->remove(timer_handle);
+  }
+
+  if (timer_is_valid) {
+    cancelTransformableRequest(request_handle);
```

Comment:
This line introduced a compiler warning: https://ci.ros2.org/view/packaging/job/packaging_linux/1559/warnings23Result/new/

@jacobperron Please address this.

Reply:
See #148